tests/csv_upload_tests.py

```diff
@@ -241,16 +279,29 @@ def test_import_csv(setup_csv_upload, create_csv_files):
     # make sure that john and empty string are replaced with None
     engine = get_upload_db().get_sqla_engine()
     data = engine.execute(f"SELECT * from {CSV_UPLOAD_TABLE}").fetchall()
-    assert data == [(None, 1, "x"), ("paul", 2, None)]
+    if utils.backend() == "hive":
+        # Be aware that hive only uses first value from the null values list
+        # it can be fixes if we will process csv file on the superset webserver.
+        assert data == [("john", 1, "x"), ("paul", 2, None)]
+    else:
+        assert data == [(None, 1, "x"), ("paul", 2, None)]
```

Comment:
I didn't quite understand this, why is "john" showing up here on Hive when other dbs return `None`?

Reply:
see the comment above, hive has it's own csv upload implementation and only supports a single null value where as default implementation works with the list. [ hive engine limitation ]

It can manually process csv to make it feature complete. Updated the comment to be more descriptive & added a todo